packages/api/amplify_api/example/integration_test/graphql/iam_test.dart

```diff
@@ -0,0 +1,371 @@
+        $listBlogs {
+          $items {
+          \tid
+          \tname
+          }
+        }
+      }''';
+
+        final operation = Amplify.API
+            .query<String>(request: GraphQLRequest(document: graphQLDocument));
+        final res = await operation.response;
+        Map data = jsonDecode(res.data!) as Map;
+        throwIfError(res);
+        expect(data[listBlogs][items], hasLength(greaterThanOrEqualTo(0)));
+      });
+
+      // Queries
+      testWidgets('should GET a blog with Model helper',
+          (WidgetTester tester) async {
+        String name = 'Integration Test Blog to fetch';
+        Blog blog = await addBlog(name);
+        final req = ModelQueries.get(Blog.classType, blog.id);
+        final res = await Amplify.API.query(request: req).response;
+        Blog? data = res.data;
+        throwIfError(res);
+        expect(data, equals(blog));
+      });
+
+      testWidgets('should LIST blogs with Model helper',
+          (WidgetTester tester) async {
+        String blog1Name = 'Integration Test Blog 1';
+        String blog2Name = 'Integration Test Blog 2';
+        String blog3Name = 'Integration Test Blog 3';
+        Blog blog1 = await addBlog(blog1Name);
+        Blog blog2 = await addBlog(blog2Name);
+        Blog blog3 = await addBlog(blog3Name);
+        final req = ModelQueries.list<Blog>(Blog.classType);
+        final operation = Amplify.API.query(request: req);
+        final res = await operation.response;
+        final data = res.data;
+        final blogs = [blog1, blog2, blog3];
+        throwIfError(res);
+        expect(data?.items, containsAll(blogs));
+      });
+
+      testWidgets(
+          'get requestForNextResult should produce next page of results from first response',
+          (WidgetTester tester) async {
+        const limit = 1;
+        final firstReq = ModelQueries.list<Blog>(Blog.classType, limit: limit);
+        final firstRes = await Amplify.API.query(request: firstReq).response;
+        final firstData = firstRes.data;
+        expect(firstData?.items.length, limit);
+        expect(firstData?.hasNextResult, true);
+        final secondReq = firstData?.requestForNextResult;
+        final secondRes = await Amplify.API.query(request: secondReq!).response;
+        final secondData = secondRes.data;
+        expect(secondData?.items.length, limit);
+        final firstId = firstData?.items[0]?.id;
+        final secondId = secondData?.items[0]?.id;
+        expect(firstId?.isNotEmpty, isTrue);
+        expect(secondId?.isNotEmpty, isTrue);
+        expect(firstId, isNot(secondId));
+      });
+
+      testWidgets('should LIST blogs with Model helper with query predicate',
+          (WidgetTester tester) async {
+        String blogName = 'Integration Test Blog ${uuid()}';
+        Blog blog = await addBlog(blogName);
+
+        final req = ModelQueries.list<Blog>(Blog.classType,
+            where: Blog.NAME.eq(blogName) & Blog.ID.eq(blog.id));
```

Comment:
*nit:* Doesn't have to be part of this PR, but we should get the lint rule that enforces trailing commas enabled in the branch/package.

This is so much easier to read imo

```dart
final req = ModelQueries.list<Blog>(
  Blog.classType,
  where: Blog.NAME.eq(blogName) & Blog.ID.eq(blog.id),
);
```

Reply:
agree, there is a task for this in asana as it's hard to enforce manually